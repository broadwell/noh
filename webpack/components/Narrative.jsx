import React from "react";
import PropTypes from "prop-types";
import { Markup } from "interweave";

// This component needs to take a string of HTML, parse it,
// determine if it has tabs, and render accordingly

const Narrative = props => (
  <div className="narrative">
    <Markup content={props.narrative} />
  </div>
);

Narrative.propTypes = {
  narrative: PropTypes.string
};

Narrative.defaultProps = {
  narrative: ""
};

export default Narrative;
