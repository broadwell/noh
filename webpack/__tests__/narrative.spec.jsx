import React from "react";
import { shallow } from "enzyme";
import Narrative from "../components/Narrative";

describe("<Narrative>", () => {
  const tabbedAnalysis =
    "<div title='section one with many words in title' class='tabbed-narrative'><p>Lorem ipsum dolor sit amet.</p></div><br><div title='section two'><p>Different stuff</p></div>";

  const nonTabbedAnalysis =
    "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam.";

  it("renders as expected without tabs", () => {
    const component = shallow(<Narrative narrative={nonTabbedAnalysis} />);
    expect(component).toMatchSnapshot();
  });

  it("renders as expected with tabs", () => {
    const component = shallow(<Narrative narrtive={tabbedAnalysis} />);
    expect(component).toMatchSnapshot();
  });
});
