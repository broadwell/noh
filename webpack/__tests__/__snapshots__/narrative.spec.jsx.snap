// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Narrative> renders as expected with tabs 1`] = `
<div
  className="narrative"
>
  <NonTabbedNarrative
    narrative=""
  />
</div>
`;

exports[`<Narrative> renders as expected without tabs 1`] = `
<div
  className="narrative"
>
  <NonTabbedNarrative
    narrative="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam."
  />
</div>
`;
